Complex navigator display fixing

jmedinaebi opened the pull request.

susiehgt committed:
Fixing of the lack of stickiness from side panel when subcomplex expanded;
Fixing of the rating display and adding to a waiting time before display;
Making the complexNavigator component adjusting to the screen;

susiehgt committed:
Fixing on display issues for Firefox
Retrieving of the complexes data for the basket
Making the complex navigator more responsive for the size of the display

susiehgt committed:
Fixing on display issues for Firefox
Retrieving of the complexes data for the basket
Making the complex navigator more responsive for the size of the display

susiehgt committed:
Fixing basket Complex Navigator display

susiehgt committed:
Merge remote-tracking branch 'origin/complexNavigatorDisplayFixing' into complexNavigatorDisplayFixing

# Conflicts:
#	src/app/basket/basket.component.html
#	src/app/basket/basket.component.ts
#	src/app/basket/basket.module.ts
#	src/app/complex/complex-results/complex-navigator/complex-navigator.component.css
#	src/app/complex/complex-results/complex-navigator/complex-navigator.component.html
#	src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.css
#	src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-stoichiometry/table-interactor-stoichiometry.component.css
#	src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-main-interactor/table-main-interactor.component.css

susiehgt committed:
Fixing basket Complex Navigator display;
Fixing Safari and Firefox display
Fixing of the rating notification
Changing Participants to Components in the complex navigator and complex viewer

susiehgt committed:
Reloading of the complex navigator in basket when element removed
Removing of comments

susiehgt committed:
Starting of classifyComplexesSimilaritiesV2 function;
display more adaptative but messing up with list view;
compact mode back to its previous state;

jmedinaebi commented on src/app/basket/basket.component.css:
```diff
@@ -0,0 +1,5 @@
+.ComplexNavigator {
+  float: left;
+  width: 100%
+  /*background-color: #6dab49;*/
```
Remove this

jmedinaebi commented on src/app/basket/basket.component.ts:
```diff
@@ -11,13 +15,18 @@ import {Title} from '@angular/platform-browser';
+  private _query: string;
+  complexSearchBasket: ComplexSearchResult = null;
+  allInteractorsInComplexSearchBasket: Interactor[] = [];
+  private _complexes: Element[];
```
Not used, remove.

jmedinaebi commented on src/app/basket/basket.component.ts:
```diff
@@ -45,4 +55,51 @@ export class BasketComponent implements OnInit, AfterViewInit {
+    this.allInteractorsInComplexSearchBasket = [];
+    this._complexes = [];
+    this.requestComplexesForNavigator();
+    console.log(this._complexes);
```
Remove console.log calls

jmedinaebi commented on src/app/basket/basket.component.html:
```diff
@@ -13,13 +13,13 @@ <h1>Personal complex collection</h1>
       <tbody>
       <tr *ngFor="let key of getKeys(complexBasket)">
```
Should we get rid of the list? Or add list/navigator buttons, similar to the ones in the search?

jmedinaebi commented on src/app/complex/complex-details/complex-participants/complex-participant-legend/complex-participant-legend.component.html:
```diff
@@ -1,58 +1,58 @@
+  <tr class="text-center small-row" *ngIf="participant.participants.length > 0">
+    <td style="{{getBackgroundStyle()}}"></td>
+    <td colspan="2" *ngIf="participant.expanded" class="no-padding-top">
+      <a class="label" (click)="participant.expanded = false">Hide components</a>
```
Should we keep this as subcomponents?

jmedinaebi commented on src/app/complex/complex-results/complex-navigator-buttons/complex-navigator-buttons.component.css:
```diff
@@ -16,7 +16,6 @@
 }
 
 .filters {
-  width: 120.5%;
```
Why was this removed?

susiehgt replied:
As we moved the complex navigator into a div to be modified depending on the basket and the search result page, the buttons were not centered anymore with this width

jmedinaebi commented on src/app/complex/complex-results/complex-navigator-buttons/complex-navigator-buttons.component.css:
```diff
@@ -33,6 +32,10 @@
   display: none;
 }
 
+label {
```
Why do we need this?

susiehgt replied:
We made some modification for the menus so the user can click directly on the whole tile rectangle instead of clicking on the radiobutton, to do so we encapsulated it in the label which needed some style adjustments

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts:
```diff
@@ -204,59 +208,107 @@ export class TableInteractorColumnComponent implements OnChanges {
         if (!!match) {
-          if (!!match.stochiometryValue) {
-            oneInteractor.timesAppearing += match.stochiometryValue[0];
+          oneInteractor.timesAppearing += 1;
```
`oneInteractor.timesAppearing += 1;` is called in this method and the method below, so we're setting `timesAppearing` to double the real value.
 It doesn't matter much, as all interactors have their number duplicated, so they are still in order, but to fix it we have 2 options.

1. Merge this 2 methods into 1, that does it all. They both have the same for loops, so it will just moving the if/else from one method to another.

2. Keep these 2 methods, without the `oneInteractor.timesAppearing += 1;` line, and have another method, `calculateTimesAppearing()` that only does that. Then call the 3 method on the `onChanges` method.

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-main-interactor/table-main-interactor.component.css:
```diff
@@ -1,11 +1,26 @@
+  align-content: center;
+  flex-wrap: nowrap;
+
+  * {
```
What's `*`?

susiehgt replied:
it selects all the elements contained in :host to apply the flex on them at the same rate, it's there for the firefox and safari compatibility

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts:
```diff
@@ -20,7 +20,7 @@ export class TableStructureComponent implements OnChanges {
   ngOnChanges(): void {
     this.classifyComplexesSize(); // ensure to always have the same base of comparison
-    this.sortedComplexes = this.classifyComplexesSimilarities(this.complexSearch.elements);
+    this.sortedComplexes = this.classifyComplexesSimilaritiesV2(this.complexSearch.elements);
```
What has changed here?

susiehgt replied:
the classifyComplexesSimilaritiesV2 is the function supposed to cluster the complexes but it is not finished yet, right now it is only classifying the complexes depending on the similarity score they obtain

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-name/table-interactor-name.component.html:
```diff
@@ -19,26 +19,19 @@
-           [queryParams]="{query: interactorId, page: 1}"
-           target="_blank">
+      <div class="name" title="More information about containing {{interactorName}}" *ngIf="!IDDisplay">
+        <a href="{{identifierLink}}" target="_blank" class="externalLinkContainer">
```
Add `*ngIf="!!identifierLink"` to this `<a>` element.

jmedinaebi commented:
I don't know if the code in the github pages is up to date, but I just tried and the search results look wonky.

For https://complex-portal.github.io/complex-portal-view/complex/search?query=%22nuclear%20pore%22#view_list

![Screenshot 2024-06-20 at 13 25 30](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/123e701a-3cc6-45e3-8659-ed38ddab6532)
![Screenshot 2024-06-20 at 13 25 36](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/0d9e5646-2667-43a1-b6ed-4b0fc5723140)

jmedinaebi commented:
I see a lot of CSS changes, are those for the Safari and Firefox support?

jmedinaebi commented on src/app/complex/complex-results/complex-navigator-buttons/complex-navigator-buttons.component.html:
```diff
@@ -2,12 +2,12 @@
   <div class="sortingInteractors">
     <input type="submit" name="classificationButton" value="Sort interactors" class="button">
```
Sort components

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.html:
```diff
@@ -28,7 +28,7 @@
           <!-- Icon for the expandable -->
           <div class="container" *ngIf="interactor.isSubComplex">
             <a title="{{ interactor.interactor.name }} interactors" (click)="toggleSubcomplexExpandable(i)">
```
`<a title="{{ interactor.interactor.name }} interactors"...` should be `<a title="{{ interactor.interactor.name }} components" ...`

jmedinaebi replied:
Same in lines 58 and 61

jmedinaebi commented on src/app/complex/complex-details/complex-participants/complex-participants.component.html:
```diff
@@ -1,5 +1,5 @@
 <div class="row" *ngIf="participantsWithSubcomponents">
-  <h2 class="float-left">Participants</h2>
+  <h2 class="float-left">Components</h2>
```
Since we change this, we should also change it in `go-to.component.html`, line 9, from Participants to Components.

jmedinaebi replied:
Then, in `complex-filter.component.html`, should it be `Interactor Type` or `Component Type`? Can you check that with Sucharita?

jmedinaebi replied:
As Sucharita said, let's rename this as Component Type.

susiehgt committed:
Complex Nav adaptive screen,
classification of the complexes by similarity score seems ok? <- to see,
Replacing interactor for component

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts:
```diff
@@ -327,4 +378,19 @@ export class TableInteractorColumnComponent implements OnChanges {
+      for (const complex of this.complexes) {
+        const match = findInteractorInComplex(complex, oneInteractor.interactor.identifier, this.enrichedInteractors);
+        if (!!match) {
+          if (!!match.stochiometryValue) {
+            oneInteractor.timesAppearing += match.stochiometryValue[0];
+          } else {
+            oneInteractor.timesAppearing += 1;
+          }
```
We could simply replace these 5 lines with
```
oneInteractor.timesAppearing += 1;
```

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts:
```diff
@@ -100,14 +102,16 @@ export class TableStructureComponent implements OnChanges {
   }
 
-  private classifyComplexesSimilarities(bigComplexes: Element[]): Element[] {
+  private classifyComplexesSimilarities(complexesList: Element[]): Element[] {
```
We can ger rid of this method, right?

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts:
```diff
@@ -132,4 +136,59 @@ export class TableStructureComponent implements OnChanges {
+    return finalList;
+  }
+
+  uniqueComplexesListOrderedBySimilarity(complexesListSimilarities: [Element, Element, number][]) {
```
We have 2 for loops here where we iterate through the complexesListSimilarities list. It makes sense but it can be more slightly more efficient.

The second time we iterate we can start from where we are, as anything before that point has already been checked. To do that, we can use indexes in the for loop.

```
    for (let i = 0; i < complexesListSimilarities.length; i++) {
      const complex1 = complexesListSimilarities[i][0];
      const complex2 = complexesListSimilarities[i][1];
      const similarityScore = complexesListSimilarities[i][2];
      if (similarityScore !== 0) {
        if (!complexesOrderedSet.has(complex1)) {
          complexesOrderedSet.add(complex1);
        }
        if (!complexesOrderedSet.has(complex2)) {
          complexesOrderedSet.add(complex2);
        }
        for (let j = i + 1; j < complexesListSimilarities.length; j++) {
          const complex3 = complexesListSimilarities[j][0];
          const complex4 = complexesListSimilarities[j][1];
          const similarityScore2nd = complexesListSimilarities[j][2];
          if (complex1 === complex3 && similarityScore2nd !== 0) {
            if (!complexesOrderedSet.has(complex4)) {
              complexesOrderedSet.add(complex4);
            }
          }
        }
      }
    }
```

I think that would make this from O(n^2) to O(nLogn), and it should work the same way.

jmedinaebi replied:
Basically, if we have 4 complexes (CPX1, CPX2, CPX3 and CPX4), `complexesListSimilarities` could look like
```
complexesListSimilarities = [
  [CPX1, CPX1, 5],
  [CPX2, CPX2, 5],
  [CPX1, CPX2, 4],
  [CPX3, CPX3, 3],
  [CPX1, CPX4, 2],
  [CPX4, CPX4, 2],
  [CPX2, CPX3, 1],
  [CPX2, CPX4, 1],
  [CPX3, CPX4, 0],
  [CPX1, CPX3, 0]
]
```

First, on the first loop, with i=0, we check `[CPX1, CPX1, 5]` and we add CPX1 to the set.
Then, we do the second loop from j=1 (i+1), to j=length-1, and we add CPX2 and CPX4 to the set.

Then, i=1, CPX2 is already added to the set. The second loop now starts from j=2.
CPX3 is added to the set.

The rest of iterations on the loops will not add anything, so we end up with [CPX1, CPX2, CPX4, CPX3].

jmedinaebi commented:
You have changed the interactor(s)/participant(s) labels to component(s) everywhere but `go-to.component.html`, which you haven't modified, so it makes sense anyway. Can you replace Participants with Components there in line 9?

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts:
```diff
@@ -132,4 +136,59 @@ export class TableStructureComponent implements OnChanges {
+            if (!complexesOrderedSet.has(complex4)) {
+              complexesOrderedSet.add(complex4);
+            }
+          }
```
Here we could add
```
if (complexesOrderedSet.size === this.complexSearch.elements.length) {
          // All complexes have been added, we can return and stop the loops
          return complexesOrderedSet;
        }
```
To return when we already have all the complexes, so we don't need to keep iterating.

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts:
```diff
@@ -132,4 +136,59 @@ export class TableStructureComponent implements OnChanges {
+  }
+
+  uniqueComplexesListOrderedBySimilarity(complexesListSimilarities: [Element, Element, number][]) {
+    const complexesOrderedSet = new Set<Element>();
```
Is `Set` sorted? Set is definitely unique, being a set, but it is not necessarily sorted. What we want is to iterate through the elements in this set (line 165) in the order they were inserted into the set. Can you check if that is happening?

susiehgt replied:
comparedComplexes is sorting the comparedComplexes Array by the highest similarity score before creating the complexesOrderedSet using uniqueComplexesListOrderedBySimilarity,  isn't it enough?

jmedinaebi replied:
We sort `complexesListSimilarities` before this method, so when we do `for (const [complex1, complex2, similarityScore] of complexesListSimilarities)` we are iterating through them in order.

Based on that, we insert into `comparedComplexes`, also in order. But `comparedComplexes` is a Set, and a set guarantees objects are unique and quick to access, but it does not guarantee it's in order of insertion.

So, when we do `for (const complex of complexesOrderedSet) {` (line 165), we are not guaranteed to iterate through the complexes in order.

We have 2 options:

1. Find an implementation of set that it is ordered. If we find one, we also don't need checks like the following, as set already doesn't allow duplicates.
    ```
    if (!complexesOrderedSet.has(complex1)) {
      complexesOrderedSet.add(complex1);
    }
    ```

2. Use a list. In line 172, we can do `const complexesOrderedSet: Element[] = [];` and it would work without any other changes. If we do this, we also don't need `finalList`, as we already have a list.

jmedinaebi replied:
I had a look at this, and it looks like Set returns the items ordered by insertion, in which case, using a set is fine. Just a few changes.

1. We don't need to call `has`, so instead of
   ```
   if (!complexesOrderedSet.has(complex1)) {
      complexesOrderedSet.add(complex1);
   }
   ```
   You can just do
   ```
   complexesOrderedSet.add(complex1);
   ```

2. Simply making the set into a list. Instead of
   ```
    for (const complex of complexesOrderedSet) {
      finalList.push(complex);
    }
    return finalList;
   ```
   You can do
   ```
   return Array.from(complexesOrderedSet);
   ```

3. Remove the commented out code.

susiehgt committed:
Fixing of list display,
removing of useless functions

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts:
```diff
@@ -204,59 +209,105 @@ export class TableInteractorColumnComponent implements OnChanges {
+    this.ranges = [];
+  }
+
+  private calculateTimesAppearingType() {
```
A couple of improvements on these methods. First, we could merge all 3 into a single method, to only iterate through interactors and components once. Second, we could simplify the has/get/set methods to deal with the map.

Regarding the maps
```
if (map.has(key)) {
    map.set(key, map.get(key) + 1);
} else {
    map.set(key, 1);
}
```
Is the same as
```
map.set((key, map.get(key) || 0) + 1);
```

Regarding merging the 3 methods (calculateTimesAppearingType, calculateTimesAppearingOrganism and interactorsTimesAppearing) into one, we could have
```
  private calculateTimesAppearing() {
    // Initialise times appearing by type or organism
    this._timesAppearingByType = new Map();
    this._timesAppearingByOrganism = new Map();
    for (const oneInteractor of this.enrichedInteractors) {
      // Initialise times appearing for each interactor
      oneInteractor.timesAppearing = 0;
      for (const complex of this.complexes) {
        const match = findInteractorInComplex(complex, oneInteractor.interactor.identifier, this.enrichedInteractors);
        if (!!match) {
          // Update times appearing for the interactor
          oneInteractor.timesAppearing += 1;
          // Update times appearing for the interactor type
          this._timesAppearingByType.set(
            oneInteractor.interactor.interactorType,
            (this._timesAppearingByType.get(oneInteractor.interactor.interactorType) || 0) + 1);
          // Update times appearing for the interactor organism
          this._timesAppearingByOrganism.set(
            oneInteractor.interactor.organismName,
            (this._timesAppearingByOrganism.get(oneInteractor.interactor.organismName) || 0) + 1);
        }
      }
    }
  }
```

susiehgt committed:
Changing of the complex similarity calculation function

jmedinaebi commented on src/app/basket/basket.component.ts:
```diff
@@ -45,4 +54,50 @@ export class BasketComponent implements OnInit, AfterViewInit {
   }
+
+  private createQuery(object: any): string {
+    let query = '';
+    for (const key of this.getKeys(object)) {
+      if (object[key] !== undefined) {
+        let queryPerObject = ' complex_id:';
+        queryPerObject += object[key].id;
+        query += queryPerObject;
+      }
+    }
+    return query;
```
This method can be simplified to just one line
```
return 'complex_id:' + Object.values(object).map((v: BasketItem) => v.id).join(',');
```

jmedinaebi commented on src/app/basket/basket.component.ts:
```diff
@@ -45,4 +54,50 @@ export class BasketComponent implements OnInit, AfterViewInit {
+  }
+
+  private requestComplexesForNavigator() {
+    this.complexPortalService.findComplex(this.createQuery(this._complexBasket)).subscribe(complexSearch => {
```
To fix the issue with only displaying 10 complexes in the basket, we need to set the the page size to the right value. By default it is 10.
```
const pageSize = Object.values(this._complexBasket).length;
    this.complexPortalService.findComplex(this.createQuery(this._complexBasket), [], [], [], 1, pageSize)
      .subscribe(complexSearch => {
```

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.ts:
```diff
@@ -327,4 +342,44 @@ export class TableInteractorColumnComponent implements OnChanges {
     return null;
   }
+
+  interactorsTimesAppearing() {
```
This method is no longer needed, we already increase `oneInteractor.timesAppearing` in line 372.

jmedinaebi commented on src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts:
```diff
@@ -19,117 +20,128 @@ export class TableStructureComponent implements OnChanges {
+    return components;
+  }
+
+  private calculateSimilarity(complex1: Element, complex2: Element) {
```
I like it, simple and short.

jmedinaebi commented:
The style still looks a bit weird.

This is how the list view and navigator view looks in https://www.ebi.ac.uk/complexportal/complex/search?query=%22nuclear%20pore%22&page=1#view_list

![Screenshot 2024-06-25 at 09 12 36](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/44ec1594-68c5-4422-bf87-5d5a9d2c4d57)

![Screenshot 2024-06-25 at 09 12 29](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/9fe5f7d0-98da-4429-9a6e-88594bd80cc4)

And this is how it currently looks in the github pages https://complex-portal.github.io/complex-portal-view/complex/search?query=%22nuclear%20pore%22&page=1#view_list

![Screenshot 2024-06-25 at 09 12 43](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/cd570e3c-a714-48b1-abc3-7bfdec36aa92)

![Screenshot 2024-06-25 at 09 12 49](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/18cf1a97-3715-4b0f-ac9d-44d5dea8768c)

I know we needed some changes for Firefox and Safari, but it looked better before. Can we revert some of the changes to go back closer to how it used to look? Particularly the padding on the left of the page.

jmedinaebi commented on src/app/complex/complex-results/complex-results.component.css:
```diff
@@ -36,12 +36,12 @@
 .leftAligned {
   --global-margin: calc((100vw - 80rem) / -2);
-  --min-margin-left: calc(-100rem * 0.17);
+  --min-margin-left: calc(-100rem * 1.17);
```
Try changing this back to how it was `--min-margin-left: calc(-100rem * 0.17);` to align the filters properly.

jmedinaebi replied:
Maybe it doesn't need to change to 1.17 again, but something in the middle. At the moment it's too close to the left border, almost as if there is no margin at all.

jmedinaebi commented on src/app/complex/complex-results/complex-results.component.css:
```diff
@@ -51,30 +51,42 @@
+
+.listOfResults {
+  width: 40vw;
+  margin-left: 17vw;
```
If you change this to 10vw, it may look a bit better `margin-left: 17vw;`

jmedinaebi replied:
After this and the other CSS change mentioned above, try in Chrome, Safari and Firefox to check everything looks ok.

jmedinaebi commented:
Another issue, when on a big screen
![Screenshot 2024-06-25 at 09 12 43](https://github.com/Complex-Portal/complex-portal-view/assets/114929382/baf9b6b5-6cf9-45a6-ad0a-40be6a8b3414)

It does not happen if I have the window on my laptop screen, but on the big screen it does.

jmedinaebi committed:
Improve navigator integration in basket

jmedinaebi committed:
Remove duplicated method

jmedinaebi committed:
Small improvements in complex ordering

jmedinaebi committed:
No more personal complexes

susiehgt committed:
Merge pull request #219 from Complex-Portal/complexNavigatorBasketImprovements

Complex navigator basket improvements

susiehgt committed:
Changing of the complex similarity calculation function

susiehgt committed:
Merge remote-tracking branch 'origin/complexNavigatorDisplayFixing' into complexNavigatorDisplayFixing

jmedinaebi committed:
Small style changes

jmedinaebi committed:
Fix issue with redirect and searches

jmedinaebi committed:
Fix table width when sorting

jmedinaebi committed:
Center Components header vertically

jmedinaebi committed:
Move basket icon

jmedinaebi committed:
Update icon used to delete complexes from basket

jmedinaebi committed:
Fixed width for small complex navigator in basket

jmedinaebi committed:
Fix query to fetch complexes in basket

jmedinaebi committed:
Fix bug with redirect on search with 1 result

susiehgt committed:
Testing for the different orthologs grouper

susiehgt committed:
Merge pull request #221 from Complex-Portal/complexNavigator-smallFixes

Complex navigator small fixes

susiehgt committed:
Merge remote-tracking branch 'origin/complexNavigatorDisplayFixing' into complexNavigatorDisplayFixing

susiehgt committed:
Remove of useless comments;
Merge with display fixing;
Starting to work on groupings

susiehgt committed:
Fixing icons for adding and removal from basket;
Working on PANTHER orthology

susiehgt committed:
Fixing icons for adding and removal from basket;
Removal of orthology addings

susiehgt committed:
Fixing icons for adding and removal from basket;
Removal of orthology addings

susiehgt committed:
Fixing icons for adding and removal from basket;
Removal of orthology addings

susiehgt committed:
Fix table width issues

susiehgt committed:
Remove of useless console log

jmedinaebi committed:
Merge branch 'develop' into complexNavigatorDisplayFixing

jmedinaebi committed:
Fix width on large table

susiehgt committed:
Adding of the remove all from the basket button;
Changing of the deletion icons in the basket;

susiehgt committed:
Changing of some basket icons

susiehgt committed:
Changing of some basket icons

susiehgt committed:
basket remove all button width fix

EliotRagueneau committed:
:arrow_up: Upgrade dependency to be able to npm install + remove unused primeng

EliotRagueneau committed:
:lipstick: Improve species & filters style

EliotRagueneau committed:
:recycle: Simplify models into interfaces + add predicted attribute

EliotRagueneau committed:
:sparkles: Use variable for primary color, in order to switch them on predicted

EliotRagueneau committed:
:bug: Use interfaces of model instead of class constructors

EliotRagueneau committed:
:lipstick: Use predicated class when needed

EliotRagueneau committed:
:lipstick: Use new basket icons

EliotRagueneau committed:
:speech_balloon: Imrpove notification grammar

EliotRagueneau committed:
:construction: MOCK predicted complexes with 50% chance

EliotRagueneau committed:
:adhesive_bandage: Restore gxa and slightly improve it

EliotRagueneau committed:
:art: Improve Legend of complexViewer

EliotRagueneau committed:
:lipstick: Improve complex header appearence + support new eco codes

EliotRagueneau committed:
:sparkles: Avoid displaying function when none is available

EliotRagueneau committed:
:sparkles: Adjust Curated by message when predicted complex

EliotRagueneau committed:
:bug: Module import for new pipe

EliotRagueneau committed:
:wastebasket: remove random predicted to be deployable

EliotRagueneau committed:
:bug: Fix CPX-5605 nested colour legend since it has been fixed in ComplexViewer 2.2.2

EliotRagueneau committed:
:bug: Revert to initial human GXA experiment

EliotRagueneau committed:
:bug: Update to new version of human experiment + support any species from Expression Atlas + poxy cof to make GXA work locally

EliotRagueneau committed:
:sparkles: Remove exceptions to rely solely on auto reference

EliotRagueneau committed:
:wastebasket: remove random predicted to be deployable

EliotRagueneau committed:
:bug: Fix missing first complex
